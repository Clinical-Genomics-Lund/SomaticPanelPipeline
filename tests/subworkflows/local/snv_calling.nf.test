nextflow_workflow {

    name "Test Workflow SNV_CALLING"
    script "subworkflows/local/snv_calling.nf"
    workflow "SNV_CALLING"

    setup {
        
        run("CHECK_INPUT") {
            script "$projectDir/subworkflows/local/create_meta.nf"
            workflow {
                """
                input[0] = file("$projectDir/tests-data/gmslymphomav3-0/23KF00457-lymf_gmslymphomav3-0.csv")
                """
            }
        }

        run("SAMPLE") {
            script "$projectDir/subworkflows/local/sample.nf"
            workflow {
                """
                input[0] = CHECK_INPUT.out.fastq
                """
            }
        }

        run("ALIGN_SENTIEON") {
            script "$projectDir/subworkflows/local/align_sentieon.nf"
            workflow {
                """
                input[0] =  SAMPLE.out.fastq_trim
                input[1] =  CHECK_INPUT.out.meta
                """
            }
        }

        run("QC") {
            script "$projectDir/subworkflows/local/qc.nf"
            workflow {
                """
                input[0] = ALIGN_SENTIEON.out.qc_out
                input[1] = ALIGN_SENTIEON.out.bam_lowcov
                """
            }
        }

    }


    test("Should run without failures") {

        when {
            
            params {
                // define parameters here. Example:
                regions_bed = "/fs1/resources/ref/hg38/bed/lymphoid-twist/v3/lymph_v3_mergedprobes_simple.bed"
            }
            
            workflow {
                """
                Channel
                    .fromPath("${params.regions_bed}")
                    .ifEmpty { exit 1, "Regions bed file not found: ${params.regions_bed}" }
                    .splitText( by: 1000, file: 'bedpart.bed' )
                    .set { beds }
                
                // define inputs of the workflow here. Example:
                input[0] = ALIGN_SENTIEON.out.bam_umi.groupTuple()
                input[1] = ALIGN_SENTIEON.out.bam_dedup
                input[2] = beds                
                input[3] = CHECK_INPUT.out.meta
                input[4] = QC.out.melt_qc
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
