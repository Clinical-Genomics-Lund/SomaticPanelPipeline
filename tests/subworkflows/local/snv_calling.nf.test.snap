{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        "23KF00457-lymf",
                        "freebayes",
                        "23KF00457-lymf_freebayes.vcf.gz:md5,cbaef071f81ada81985a128b64fbcdd9"
                    ],
                    [
                        "23KF00457-lymf",
                        "vardict",
                        "23KF00457-lymf_vardict.vcf.gz:md5,0c1b72d34b3679885c15a2d1876b44bc"
                    ]
                ],
                "1": [
                    [
                        "23KF00457-lymf",
                        [
                            {
                                "id": "23KF00457-lymf",
                                "group": "23KF00457-lymf",
                                "diagnosis": "other",
                                "type": "T",
                                "clarity_sample_id": "CMD707A3447",
                                "ffpe": "FFPE",
                                "purity": false,
                                "sequencing_run": "231024_A01932_0295_AHK5TYDRX3",
                                "reads": "75970669",
                                "clarity_pool_id": "122-355931"
                            },
                            {
                                "id": "23HL02304-lymf",
                                "group": "23KF00457-lymf",
                                "diagnosis": "other",
                                "type": "N",
                                "clarity_sample_id": "CMD707A2651",
                                "ffpe": "",
                                "purity": false,
                                "sequencing_run": "230829_A01932_0262_AHJ75FDRX3",
                                "reads": "33966764",
                                "clarity_pool_id": "122-341042"
                            }
                        ],
                        "23KF00457-lymf.agg.vcf:md5,92846b12a078ab159ebb7d485be905c5"
                    ]
                ],
                "2": [
                    "versions.yml:md5,6ea93ef784f9862e2b15291a5645ef87",
                    "versions.yml:md5,8ed730ff20acc48e51c2f3d27aa55418",
                    "versions.yml:md5,aeec1d912f15cf291c8f14eb67bae9cb",
                    "versions.yml:md5,d99037b14b59387f8e1cfdd5d85b3d78"
                ],
                "agg_vcf": [
                    [
                        "23KF00457-lymf",
                        [
                            {
                                "id": "23KF00457-lymf",
                                "group": "23KF00457-lymf",
                                "diagnosis": "other",
                                "type": "T",
                                "clarity_sample_id": "CMD707A3447",
                                "ffpe": "FFPE",
                                "purity": false,
                                "sequencing_run": "231024_A01932_0295_AHK5TYDRX3",
                                "reads": "75970669",
                                "clarity_pool_id": "122-355931"
                            },
                            {
                                "id": "23HL02304-lymf",
                                "group": "23KF00457-lymf",
                                "diagnosis": "other",
                                "type": "N",
                                "clarity_sample_id": "CMD707A2651",
                                "ffpe": "",
                                "purity": false,
                                "sequencing_run": "230829_A01932_0262_AHJ75FDRX3",
                                "reads": "33966764",
                                "clarity_pool_id": "122-341042"
                            }
                        ],
                        "23KF00457-lymf.agg.vcf:md5,92846b12a078ab159ebb7d485be905c5"
                    ]
                ],
                "concat_vcfs": [
                    [
                        "23KF00457-lymf",
                        "freebayes",
                        "23KF00457-lymf_freebayes.vcf.gz:md5,cbaef071f81ada81985a128b64fbcdd9"
                    ],
                    [
                        "23KF00457-lymf",
                        "vardict",
                        "23KF00457-lymf_vardict.vcf.gz:md5,0c1b72d34b3679885c15a2d1876b44bc"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6ea93ef784f9862e2b15291a5645ef87",
                    "versions.yml:md5,8ed730ff20acc48e51c2f3d27aa55418",
                    "versions.yml:md5,aeec1d912f15cf291c8f14eb67bae9cb",
                    "versions.yml:md5,d99037b14b59387f8e1cfdd5d85b3d78"
                ]
            }
        ],
        "timestamp": "2023-12-27T14:53:17.371132"
    }
}